devito/passes/iet/parpragma.py

```diff
@@ -307,7 +307,7 @@ def _make_nested_partree(self, partree):
             # within a block)
             candidates = []
             for i in inner:
-                if self.key(i) and any(is_integer(j.step-i.symbolic_size) for j in outer):
+                if self.key(i) and any((j.dim.parent is i.dim.root) for j in outer):
```

Comment:
I'm not sure this will work if we have SubDimensions

could you add a nested parallelism over SubDimensions test here: https://github.com/devitocodes/devito/blob/master/tests/test_dle.py#L620

probably using `grid.interior` suffices